packages/nextra/src/loader.ts

```diff
@@ -211,6 +210,14 @@ const Content = props => (
   </__nextra_SSGContext__.Provider>
 )
 
+try {
+  if (getStaticPaths) {
+    const temp = getStaticPaths
+    getStaticPaths = () => temp(__nextra_pageOpts__)
+  }
+} catch {
+}
```

Comment:
```suggestion
if (typeof getStaticPaths !== 'undefined') {
  const temp = getStaticPaths
  getStaticPaths = () => temp(__nextra_pageOpts__)
}
```
Silently ignore errors it's not good in this case

Reply:
also, this piece of code can be removed since getStaticPaths needs `pageMap` that can be retrieved by `globalThis.__nextra_internal__.pageMap`

Reply:
My concern is that exporting a const won't work:

```js
export const x = 1

x = 2
```

Reply:
I use this pattern beacuse i dont want touch `globalThis`. 

The i realize i could access `__nextra_pageOpts__` in mdx directly

```mdx
import { useSSG } from 'nextra/ssg'
import { getStaticTags } from 'nextra-theme-blog'

export const TagName = () => {
  const { tag } = useSSG()
  return tag || null
}

# Posts Tagged with “<TagName/>”

export const getStaticPaths = () => {
  const tags = getStaticTags(__nextra_pageOpts__)
  return {
    paths: tags.map(v => ({ params: { tag: v } })),
    fallback: false
  }
}
export const getStaticProps = ({ params: { tag } }) => {
  return {
    props: {
      ssg: {
        tag
      }
    }
  }
}
```